src/background/services/NativeAppService.ts

```diff
@@ -56,6 +60,7 @@ export default class NativeAppService {
 
       if (message.version) {
         this.state = NativeAppState.CONNECTED;
+        new Promise((resolve, reject) => this.activeConnection = { resolve, reject });
```

Comment:
What a curious trick :)! But cheers, I see that `await Promise.race([this.activeConnection, throwAfterTimeout(...)])` below would not be possible without this.